src/Behaviors/Meshes/baseSixDofDragBehavior.ts

```diff
@@ -0,0 +1,378 @@
+import { Behavior } from "../../Behaviors/behavior";
+import { Mesh } from "../../Meshes/mesh";
+import { AbstractMesh } from "../../Meshes/abstractMesh";
+import { Scene } from "../../scene";
+import { Nullable } from "../../types";
+import { PointerInfo, PointerEventTypes } from "../../Events/pointerEvents";
+import { Vector3, Quaternion } from "../../Maths/math.vector";
+import { Observer, Observable } from "../../Misc/observable";
+import { TransformNode } from "../../Meshes/transformNode";
+import { Camera } from "../../Cameras/camera";
+
+/**
+ * Base behavior for six degrees of freedom interactions in XR experiences.
+ * Creates virtual meshes that are dragged around
+ * And observables for position/rotation changes
+ */
+export class BaseSixDofDragBehavior implements Behavior<Mesh> {
+    protected static _virtualScene: Scene;
+    private _pointerObserver: Nullable<Observer<PointerInfo>>;
+    private _attachedToElement: boolean = false;
+    protected _virtualMeshesInfo: {

```

Comment:
Couldn't find any dispose function in this class. Is detach considered to be dispose in this case?

Reply:
Yes, all behaviors have a `detach` function but no `dispose` in the interface